hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/contracts/HederaEvmStackedWorldStateUpdater.java

```diff
@@ -16,10 +16,30 @@
 package com.hedera.node.app.service.evm.store.contracts;
 
 import com.hedera.node.app.service.evm.accounts.AccountAccessor;
+import com.hedera.node.app.service.evm.store.models.UpdatedHederaEvmAccount;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.account.EvmAccount;
+import org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount;
+import org.hyperledger.besu.evm.worldstate.WrappedEvmAccount;
 
 public class HederaEvmStackedWorldStateUpdater extends AbstractLedgerEvmWorldUpdater {
 
-    public HederaEvmStackedWorldStateUpdater(AccountAccessor accountAccessor) {
+    protected final HederaEvmEntityAccess hederaEvmEntityAccess;
+
+    public HederaEvmStackedWorldStateUpdater(
+            AccountAccessor accountAccessor, HederaEvmEntityAccess hederaEvmEntityAccess) {
         super(accountAccessor);
+        this.hederaEvmEntityAccess = hederaEvmEntityAccess;
+    }
+
+    @Override
+    public EvmAccount getAccount(Address address) {
+        final var balance = Wei.of(hederaEvmEntityAccess.getBalance(address));
+        final var evmAccount =
+                new UpdatedHederaEvmAccount(accountAccessor.canonicalAddress(address));
+        evmAccount.setBalance(balance);
```

Comment:
Nonce is intentionally empty, if it's needed an additional method in HederaEvmEntityAccess needs to be introduced. However all it's implementors need to provide implementation for it, and in mono-sevices this seem redundant.